sycl/test/acap/test_aie_allocator_test.cpp

```diff
@@ -0,0 +1,72 @@
+// REQUIRES: acap
+
+// RUN: %acap_clang %s -o %s.bin
+// RUN: %add_acap_result %s.bin
+// RUN: rm %s.bin
+
+#include <sycl/sycl.hpp>
+
+#include "triSYCL/vendor/Xilinx/acap/aie/device_allocator.hpp"
+
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+constexpr unsigned count = 10;
+
+using namespace sycl::vendor::xilinx;
+/// A memory tile has to inherit from acap::aie::memory<AIE, X, Y>
+template <typename AIE, int X, int Y>
+struct tile_memory : acap::aie::memory<AIE, X, Y> {
+  acap::hw::dev_ptr<void> arr[count];
+};
+
+template <typename AIE, int X, int Y> struct prog : acap::aie::tile<AIE, X, Y> {
+  using t = acap::aie::tile<AIE, X, Y>;
+  /// The lock function of every tiles is run before the prerun of any other
+  /// tile.
+  bool prerun() {
+    return 1;
+  }
```

Comment:
Which `lock`?
Why this `prerun()` API which should have disappeared long time ago?